---
bibliography: paper/library.bib
toc: false
citation-location: margin
---

![](https://raw.githubusercontent.com/funkyheatmap/logo/refs/heads/main/src/funkyheatmap_edited.png){width=256 fig-align="center"}

**{funkyheatmap}** allows generating heatmap-like visualisations for data frames. Funky heatmaps can be fine-tuned by providing annotations of the columns and rows, which allows assigning multiple palettes or geometries or grouping rows and columns together in categories.

## Implementations

* JavaScript (incomplete): [Repo](https://github.com/funkyheatmap/funkyheatmapjs), [npm](https://www.npmjs.com/package/funkyheatmapjs)
* Python: [Repo](https://github.com/funkyheatmap/funkyheatmappy), [PyPI](https://pypi.org/project/funkyheatmappy), [Docs](https://funkyheatmap.github.io/funkyheatmappy)
* R: [Repo](https://github.com/funkyheatmap/funkyheatmap), [CRAN](https://cran.r-project.org/package=funkyheatmap), [Docs](https://funkyheatmap.github.io/funkyheatmap)


## Gallery

These are works with funkyheatmap (-inspired) visualisations. Add your work to the list by creating a PR!

::: {layout-ncol=2}

![Wouter Saelens, Robrecht Cannoodt, Helena Todorov, and Yvan Saeys. 2019. “A Comparison of Single-Cell Trajectory Inference Methods.” Nature Biotechnology. doi: [10.1038/s41587-019-0071-9](https://doi.org/10.1038/s41587-019-0071-9).](images/saelens2019.png){width=320px}

![Luecken, Malte D., M. Büttner, K. Chaichoompu, A. Danese, M. Interlandi, M. F. Mueller, D. C. Strobl, et al. 2021. “Benchmarking Atlas-Level Data Integration in Single-Cell Genomics.” Nature Methods. doi: [10.1038/s41592-021-01336-8](https://doi.org/10.1038/s41592-021-01336-8).](images/luecken2021.png){width=320px}

![Mereu, Elisabetta, Atefeh Lafzi, Catia Moutinho, Christoph Ziegenhain, Davis J. McCarthy, Adrián Álvarez-Varela, Eduard Batlle, et al. 2020. “Benchmarking Single-Cell RNA-Sequencing Protocols for Cell Atlas Projects.” Nature Biotechnology. doi: [10.1038/s41587-020-0469-4](https://doi.org/10.1038/s41587-020-0469-4).](images/mereu2020.png){width=320px}

![Pratapa, Aditya, Amogh P. Jalihal, Jeffrey N. Law, Aditya Bharadwaj, and T. M. Murali. 2019. “Benchmarking Algorithms for Gene Regulatory Network Inference from Single-Cell Transcriptomic Data.” bioRxiv. doi: [10.1101/642926](https://doi.org/10.1101/642926).](images/pratapa2019.png){width=320px}

![Yan, Lulu, and Xiaoqiang Sun. 2022. “Benchmarking and Integration of Methods for Deconvoluting Spatial Transcriptomic Data.” Bioinformatics. doi: [10.1093/bioinformatics/btac805](https://doi.org/10.1093/bioinformatics/btac805).](images/yan2022.png){width=320px}

![Sang-aram, Chananchida, Robin Browaeys, Ruth Seurinck, and Yvan Saeys. 2023. “Spotless: A Reproducible Pipeline for Benchmarking Cell Type Deconvolution in Spatial Transcriptomics,” eLife. doi: [10.7554/eLife.88431.3](https://doi.org/10.7554/eLife.88431.3).](images/sangaram2024.png){width=320px}

:::
